service/k8sclient/kubectl/kubectl.go

```diff
@@ -104,15 +110,27 @@ func saveKubeconfig(kubeconfig string) (string, string, error) {
 	return tmpDir, kubeconfigPath, nil
 }
 
-// getKubectlCmd gets correct version of kubectl binary for Kubernetes cluster.
-func getKubectlCmd(ctx context.Context) ([]string, error) {
-	// Firstly lookup default kubectl to get Kubernetes Server version.
-	kubectlCmd, err := lookupCorrectKubectlCmd([]string{defaultKubectl})
-	if err != nil {
-		return nil, err
+func getDefaultKubectlCmd() ([]string, error) {
+	kubectlPath, errPmmServerKubectl := exec.LookPath(defaultPmmServerKubectl)
```

Comment:
Why can't we just use `lookupCorrectKubectlCmd` with two values in slice instead of this method?

Reply:
Yeah, I also thought in such a way and even started to implement it.
But then I refactored it to separate them, follows a rule: one task one function.
We have two steps:
1. find default kubectl;
2. find the latest valid kubectl version for server
and this makes the logic more evident if we have a dedicated function to each step.


In opposite: joining it into one function introduce an additional hidden dependency on order on the lookup list with the default and introduce additional complexity for local dev env (as it is not a real path but "alias").

Reply:
I didn't suggest to replace it to `getKubectlCmd`, I suggested to replace it to `lookupCorrectKubectlCmd`

Reply:
Fixed.